cfpb/consumerfinance.gov
# Add django-cors-headers and env var for origins

This PR adds the [django-cors-headers](https://pypi.org/project/django-cors-headers/) package to automatically add Cross-Origin Resource Sharing (CORS) headers via settings and middleware to allow origins specified in the `CORS_ALLOWED_ORIGINS` environment variable.

The intended use of this is to permit GovDelivery subscriptions from the [HMDA pages](https://ffiec.cfpb.gov/).

## How to test this PR

To test the CORS Access-Control-Allow-Origin header, set the `CORS_ALLOWED_ORIGINS` environment variable to a valid JSON array of origins:

```bash
export CORS_ALLOWED_ORIGINS=[\"https://example.com\"]
./runserver.sh
```

Then send the `Origin` header in a test request with one of those origins:

```bash
curl -I -H "Origin: https://example.com" http://localhost:8000
```

And observe the `Access-Control-Allow-Origin` header in the response with that origin:

```http
Access-Control-Allow-Origin: https://example.com
```

## Notes and todos

- To use this in deployments, this will require adding the `CORS_ALLOWED_ORIGINS` variable to our Ansible config.

## Checklist
